src/NUnitFramework/framework/Internal/Execution/WorkItem.cs

```diff
@@ -205,149 +205,49 @@ public virtual string Name
+                        ? "RequiresThreadAttribute may not be specified on a test within a single-SingleThreadedAttribute fixture."
+                        : "Tests in a single-threaded fixture may not specify a different apartment";
 
                     log.Error(msg);
                     Result.SetResult(ResultState.NotRunnable, msg);
                     WorkItemComplete();
-                }
-            }
-            else
-                RunTest();
-        }
-
-        private Thread thread;
-
-        private void RunTestOnOwnThread(int timeout, ApartmentState apartment)
-        {
-            thread = new Thread(new ThreadStart(RunTest));
-            thread.SetApartmentState(apartment);
-            RunThread(timeout);
-        }
-
-        private void RunThread(int timeout)
-        {
-            thread.CurrentCulture = Context.CurrentCulture;
-            thread.CurrentUICulture = Context.CurrentUICulture;
-            thread.Start();
-            
-            if (timeout <= 0)
-                timeout = Timeout.Infinite;
-
-            if (!thread.Join(timeout))
-            {
-                // Don't enforce timeout when debugger is attached.
-                // We perform this check after the initial timeout has passed to 
-                // give the user additional time to attach a debugger 
-                // after the test has started execution
-                if (Debugger.IsAttached)
-                {
-                    thread.Join();
                     return;
                 }
 
-                Thread tThread;
-                lock (threadLock)
-                {
-                    if (thread == null)
-                        return;
-
-                    tThread = thread;
-                    thread = null;
-                }
-
-                if (Context.ExecutionStatus == TestExecutionStatus.AbortRequested)
-                    return;
-
-                log.Debug("Killing thread {0}, which exceeded timeout", tThread.ManagedThreadId);
-                ThreadUtility.Kill(tThread);
-
-                // NOTE: Without the use of Join, there is a race condition here.
-                // The thread sets the result to Cancelled and our code below sets
-                // it to Failure. In order for the result to be shown as a failure,
-                // we need to ensure that the following code executes after the
-                // thread has terminated. There is a risk here: the test code might
-                // refuse to terminate. However, it's more important to deal with
-                // the normal rather than a pathological case.
-                tThread.Join();
-
-                log.Debug("Changing result from {0} to Timeout Failure", Result.ResultState);
-
-                Result.SetResult(ResultState.Failure,
-                    string.Format("Test exceeded Timeout value of {0}ms", timeout));
-
-                WorkItemComplete();
-            }
-        }
+#if DEBUG
```

Comment:
Is there a reason this logging is `#if DEBUG`? That doesn't seem to be something we normally do for `log.Debug`?